app/code/Magento/Wishlist/Model/WishlistManagement.php

```
@@ -0,0 +1,227 @@
+    {
+        if (empty($customerId) || !isset($customerId) || $customerId == "") {
+            $message = __('Id required');
+            $status = false;
+            $response[] = [
+                "message" => $message,
+                "status"  => $status
+            ];
+            return $response;
+        } else {
+            $collection =
+                $this->_wishlistCollectionFactory->create()
+                    ->addCustomerIdFilter($customerId);
+
+            $wishlistData = [];
+            foreach ($collection as $item) {
+                $productInfo = $item->getProduct()->toArray();
+                $data = [
+                    "wishlist_item_id" => $item->getWishlistItemId(),
+                    "wishlist_id"      => $item->getWishlistId(),
+                    "product_id"       => $item->getProductId(),
+                    "store_id"         => $item->getStoreId(),
+                    "added_at"         => $item->getAddedAt(),
+                    "description"      => $item->getDescription(),
+                    "qty"              => round($item->getQty()),
+                    "product"          => $productInfo
+                ];
+                $wishlistData[] = $data;
+            }
+            return $wishlistData;
+        }
+    }
+
+    /**
+     * Add wishlist item for the customer
+     * @param int $customerId
+     * @param int $productIdId
+     * @return array|bool
+     *
+     */
+    public function addWishlistForCustomer($customerId, $productId)
+    {
+        if ($productId == null) {
+            $message = __('Invalid product, Please select a valid product');
+            $status = false;
+            $response[] = [
+                "message" => $message,
+                "status"  => $status
+            ];
+            return $response;
+        }
+        try {
+            $product = $this->_productRepository->getById($productId);
+        } catch (Exception $e) {
+            return false;
```

Exception swallowing is a bad practice, exception should bubble up, up to the place which knows how to handle this exception